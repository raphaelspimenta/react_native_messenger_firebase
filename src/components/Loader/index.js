import React from 'react'
import { Modal, ActivityIndicator } from 'react-native'
import PropTypes from 'prop-types'
import { ModalBackground, ModalWrapper } from './styled'

const Loader = (props) => {
  const { loading } = props

  return (
    <Modal
      transparent
      animationType="none"
      visible={loading}
      onRequestClose={() => {}}
    >
      <ModalBackground>
        <ModalWrapper>
          <ActivityIndicator
            animating={loading}
          />
        </ModalWrapper>
      </ModalBackground>
    </Modal>
  )
}

Loader.propTypes = {
  loading: PropTypes.bool.isRequired,
}

export default Loader
