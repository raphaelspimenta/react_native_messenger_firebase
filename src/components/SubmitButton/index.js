import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import SubmitButtonStyled from './styled'

class SubmitButton extends PureComponent {

  render() {
    const { onPress } = this.props

    return <SubmitButtonStyled {...this.props} onPress={onPress} />
  }

}

SubmitButton.propTypes = {
  onPress: PropTypes.func.isRequired,
}

export default SubmitButton
