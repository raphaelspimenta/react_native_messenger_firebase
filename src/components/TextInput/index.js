import React, { PureComponent } from 'react'
import TextInputStyled from './styled'

class TextInput extends PureComponent {

  render() {
    return <TextInputStyled {...this.props} />
  }

}

export default TextInput
