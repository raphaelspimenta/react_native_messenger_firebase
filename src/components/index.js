import SubmitButton from './SubmitButton'
import TextInput from './TextInput'
import Loader from './Loader'

export {
  SubmitButton,
  TextInput,
  Loader,
}
