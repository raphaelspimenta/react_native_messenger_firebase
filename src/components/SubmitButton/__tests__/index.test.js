import React from 'react'
import { shallow } from 'enzyme'
import { SubmitButton } from 'components'

describe('Button Component', () => {
  it('snapshot ', () => {
    const component = shallow(<SubmitButton title="Just a button" onPress={() => {}} />)
    expect(component).toMatchSnapshot()
  })
})
