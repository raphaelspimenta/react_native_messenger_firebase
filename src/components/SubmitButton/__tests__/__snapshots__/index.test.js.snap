// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshot  1`] = `
<Styled(Button)
  onPress={[Function]}
  title="Just a button"
/>
`;
